test/hooks.test.js

```diff
@@ -1334,6 +1340,58 @@ test('Register an hook after a plugin inside a plugin (with beforeHandler)', t =
   })
 })
 
+test('Register hooks inside a plugin after an encapsulated plugin', t => {
```

Comment:
@nwoltman This test checks the hook order. The second hook preHandler is called before the first one. Why? 

Both of preHabdler hooks are been registered inside a plugin

Reply:
The "second" `preHandler` hook is called first because it is registered inside a non-encapsulated plugin (using `fp`) so it is registered on the root Fastify instance, which is the parent of the first (encapsulated) plugin. Hooks defined on a parent instance are always called before the hooks in a child instance.

It's the same as if the second `preHandler` hooks was defined in the same place, but not inside a plugin.